sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/Metrics/MetricPeriodFeedback.cs

```diff
@@ -1,16 +1,43 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System;
 using Azure.Core;
 
 namespace Azure.AI.MetricsAdvisor.Models
 {
-    /// <summary> The PeriodFeedback. </summary>
+    /// <summary>
+    /// Feedback indicating that this is an interval of seasonality.
+    /// </summary>
     [CodeGenModel("PeriodFeedback")]
     public partial class MetricPeriodFeedback : MetricFeedback
     {
+        /// <summary> Initializes a new <see cref="MetricPeriodFeedback"/> instance. </summary>
+        /// <param name="metricId"> The metric unique id. </param>
+        /// <param name="dimensionFilter"> The <see cref="FeedbackDimensionFilter"/> to apply to the feedback. </param>
+        /// <param name="value"> The <see cref="PeriodFeedbackValue"/> for the feedback. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="metricId"/>, <paramref name="dimensionFilter"/>, or <paramref name="value"/> is null. </exception>
+        public MetricPeriodFeedback(string metricId, FeedbackDimensionFilter dimensionFilter, PeriodFeedbackValue value) : base(metricId, dimensionFilter)
+        {
+            if (metricId == null)
+            {
+                throw new ArgumentNullException(nameof(metricId));
+            }
```

Comment:
We should have an empty-check for `metricId` (and update documentation accordingly).